plugins/nf-azure/src/main/nextflow/cloud/azure/batch/AzBatchService.groovy

```diff
@@ -375,12 +375,18 @@ class AzBatchService implements Closeable {
                 .trim()
                 .replaceAll(/[^a-zA-Z0-9-_]+/,'_')
 
-        final String key = "job-${Rnd.hex()}-${name}"
+        final String key = includeTowerPrefix("job-${Rnd.hex()}-${name}", System.getenv())
```

Comment:
Better using [SysEnv.get](https://github.com/nextflow-io/nextflow/blob/6900c1b46822fa9e20d3ab135b29a52cd4b3b1e4/modules/nf-commons/src/main/nextflow/SysEnv.groovy#L40-L40) that `System.getenv`. In this case it's essentially the same, but it can simplify testing in some cases

Reply:
Done